### PR TITLE
Profile and Auth alert / message

### PR DESCRIPTION
**Pull request by henryyang42**
Change the outlook in user profile
- `忘記密碼 update 完之後 Update successfully 字太小了 而且在最下面按鈕旁邊根本就看不到` -> use message
- `讓 user 跟 subjudge 可以看到自己的 user level 但是不能改` -> add readonly form
- `改user level ＋alert` -> add `onsubmit` check

### CODE REVIEW

File: users/views.py [hunk 1]
```diff
@@ -85,12 +81,12 @@ def user_profile(request, username):
                                 (request.user, username, user_level))
                     profile_user.user_level = user_level
                     profile_user.save()
-                    render_data['userlevel_message'] = 'Update successfully'
+                    render_data['userlevel_form'] = userlevel_form
+                    messages.success(request, 'Update userlevel successfully!')
```
Comment (nickelcheng) [inline users/views.py]: I prefer to use ```user level``` or ```userLevel```

**Comment by nickelcheng** [comment]
judge can update other account to admin OAQ

File: users/forms.py [hunk 2]
```diff
@@ -133,7 +133,13 @@ def save(self):
 
 class UserLevelForm(forms.ModelForm):
     """A form for updating user's userlevel."""
-    user_level = forms.ChoiceField(label='Userlevel', choices=User.USER_LEVEL_CHOICE)
+    def __init__(self, *args, **kwargs):
+        request_user = kwargs.pop('request_user')
+        super(UserLevelForm, self).__init__(*args, **kwargs)
+        # Judge can only promote a user to these levels
+        self.fields['user_level'].label = 'User Level'
+        if request_user.user_level == User.JUDGE:
```
Comment (henryyang42) [inline users/forms.py]: Change the choices dynamically according to ``request.user``

File: users/views.py [hunk 3]
```diff
@@ -74,22 +71,19 @@ def user_profile(request, username):
...
             if can_change_userlevel(request.user, profile_user):
                 if userlevel_form.is_valid(request.user):
                     user_level = userlevel_form.cleaned_data['user_level']
-                    logger.info('User %s update %s\'s user_level to %s' %
+                    logger.info('User %s update %s\'s user level to %s' %
                                 (request.user, username, user_level))
                     profile_user.user_level = user_level
                     profile_user.save()
-                    render_data['userlevel_message'] = 'Update successfully'
-                else:
```
Comment (nickelcheng) [inline users/views.py]: Is there any possibility of sending hide choices to server?
If so, this checking should not be removed.

**Comment by nickelcheng** [comment]
Please hide user email from other users except for admin.
We'll add a checking box to decide whether an user's will public or not (maybe in next release).

File: users/templates/users/profile.html [hunk 4]
```diff
@@ -57,51 +59,52 @@
...
-                    value="{{ profile_user.username }}">
+                    id="id_username" value="{{ profile_user.username }}">
                 </div>
               </div>
-            </div>
-            <div class="form-group">
-              <div class='row'>
-                <div class='col-lg-2 col-lg-offset-1'>
-                  <label class="control-label">Email</label>
+              {% if user|reveal_private_info:profile_user %}
```
Comment (henryyang42) [inline users/templates/users/profile.html]: Hide email from other user

**Comment by nickelcheng** [comment]
+1, @bruce3557 please check

File: users/forms.py [hunk 5]
```diff
@@ -133,7 +133,13 @@ def save(self):
 
 class UserLevelForm(forms.ModelForm):
     """A form for updating user's userlevel."""
-    user_level = forms.ChoiceField(label='Userlevel', choices=User.USER_LEVEL_CHOICE)
+    def __init__(self, *args, **kwargs):
+        request_user = kwargs.pop('request_user', User())
+        super(UserLevelForm, self).__init__(*args, **kwargs)
+        # Judge can only promote a user to these levels
+        self.fields['user_level'].label = 'User Level'
+        if request_user.user_level == User.JUDGE:
```
Comment (bruce3557) [inline users/forms.py]: Is it possible to check user level in a function?
For ex: ```check_judge_user```

Comment (bruce3557) [inline users/forms.py]: I see there is a function ```has_judge_auth```
Is the purpose same with this line?

**Comment by bruce3557** [comment]
+1 besides the comments

File: users/forms.py (hunk 5 above)
Comment (henryyang42) [inline users/forms.py]: has_judge_auth will include admin,  but the choice should only provide to judge exclusively.

**Comment by nickelcheng** [comment]
There is no other comment. Going to merge.

File: users/forms.py (hunk 5 above)
Comment (nickelcheng) [inline users/forms.py]: oh, sorry, I didn't notice this comment >"<
Please fix it in another PR QQQ
About this comment, how about
```
if has_judge_auth(user):
    self.fields['user_level'].choices = ((User.SUB_JUDGE, 'Sub-judge'), (User.USER, 'User'))
elif has_admin_auth(user):
    self.fields['user_level'].label = 'User Level'
```

**Pull request merged**